File: Library.Core.Web/src/app/view-models/author/putAuthorView.ts
export class PutAuthorView {
  constructor(
    public id: number = 0,
    public name?: string,
    public birthday: number = 0,
    public deathday: number = 0,
    public books?: BookPutAuthorViewItem[]
  ) { }
}

export class BookPutAuthorViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public yearOfPublishing: number = 0
  ) { }
}

File: Library.Core.ViewModelLayer/ViewModels/Author/PutAuthorView.cs
using System.Collections.Generic;

namespace Library.Core.ViewModelLayer.ViewModels.Author
{
    public class PutAuthorView
    {
        public int AuthorId { get; set; }
        public string Name { get; set; }
        public int Birthday { get; set; }
        public int Deathday { get; set; }

        public ICollection<BookPutAuthorViewItem> Books { get; set; }

        public PutAuthorView()
        {
            Books = new List<BookPutAuthorViewItem>();
        }
    }

    public class BookPutAuthorViewItem
    {
        public int BookId { get; set; }
        public string Name { get; set; }
        public int YearOfPublishing { get; set; }

        public BookPutAuthorViewItem()
        {

        }
    }
}
